libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -527,6 +528,34 @@ private boolean isEvidenceVc(VcStoreItem item) throws CredentialParseException {
         return false;
     }
 
+    private boolean isNonZeroInt(JsonNode node) {
+        return node.isInt() && node.asInt() != 0;
+    }
+
+    private boolean isValidEvidence(VcStoreItem item) {
```

Comment:
There is already a function `isEvidenceVc()` could these be made more consistent or even combined?

Reply:
Yes, I've considered that. However, in the isEvidenceVC(), it only checks whether there is an int value. More importantly, the CredentialParseException error is propagated to the caller method without being caught. In the isValidEvidence(), it checks for credentials with valid evidence only. If a CredentialParseException error is encountered, it ignores that invalid credential. Therefore, I added a new method for the sake of clarify in these details.

Reply:
Hmm, I think actually the other method _should_ include the non-zero check. I think the exception difference isn't hugely important - in practice if we get an invalid credential we'll fail on CRI return, and won't reach any of these code paths anyway.